azclishell/az_completer.py

```diff
@@ -284,15 +296,20 @@ def gen_global_param_completions(self, text):
                         and not text.split()[-1].startswith('--') and \
                         param.startswith('-') and not param.startswith('--') and\
                         self.validate_completion(param, text.split()[-1], text, double=False):
-                    yield Completion(param, -len(text.split()[-1]))
```

Comment:
Instead of repeatedly calling 'text.split()', it would be cleaner and more efficient to call it once at the beginning of your method.